src/app/auth/authentication.service.ts

```diff
@@ -8,52 +7,42 @@ import { AUTH_API_URL } from '../shared/auth-api';
 
 @Injectable()
 export class AuthenticationService {
-  private authToken: string = '';
   private refreshInterval: number;
   private apiUrl: string;
   private clearTimeoutId: any;
 
-  constructor(private router: Router,
-              private broadcaster: Broadcaster,
+  constructor(private broadcaster: Broadcaster,
               @Inject(AUTH_API_URL) apiUrl: string,
               private http: Http) {
     this.apiUrl = apiUrl;
   }
 
-  isLoggedIn(): boolean {
-    let token = localStorage.getItem('auth_token');
-    if (token) {
-      this.authToken = token;
-      // refresh the token in five seconds to make sure we have expiry and a running timer - only do this first time in
-      if(!this.refreshInterval) {
-        this.setupRefreshTimer(15);
-      }
-      return true;
-    }
-    let params: any = this.getUrlParams();
-    if ('token_json' in params) {
-      let tokenJson = decodeURIComponent(params['token_json']);
-      let token = this.processTokenResponse(JSON.parse(tokenJson));
-      this.setupRefreshTimer(token.expires_in);
-      return true;
-    }
-    return false;
+  logIn(tokenParameter: string): boolean {
+    let tokenJson = decodeURIComponent(tokenParameter);
+    let token = this.processTokenResponse(JSON.parse(tokenJson));
+    this.setupRefreshTimer(token.expires_in);
+    this.broadcaster.broadcast('loggedin', 1);
+    return true;
   }
 
-  logout(redirect: boolean = false) {
-    this.authToken = '';
+  logout() {
     localStorage.removeItem('auth_token');
     localStorage.removeItem('refresh_token');
     clearTimeout(this.clearTimeoutId);
     this.refreshInterval = null;
     this.broadcaster.broadcast('logout', 1);
-    if (redirect) {
-      this.router.navigate(['login']);
```

Comment:
This is a big improvement, I had to create a login route in platform just to handle this!